server/plugin.go

```diff
@@ -108,18 +113,54 @@ func (p *Plugin) OnActivate() error {
 	}
 	manager.ApplyConfig(scpCfg)
 
-	sync, err := generateTargetGroup(p.API.GetConfig(), nil)
-	if err != nil {
-		return fmt.Errorf("could not set scrape target :%w", err)
-	}
-	syncCh <- sync
-
-	// check if cluster is enabled
 	if p.isHA() {
-		// TODO(isacikgoz): get cluster info
-		// we will need to push new cluster layout to p.clusterCh by either polling the cluster table
-		// or listening the cluster event messages
-		p.API.LogWarn("cluster meterics is not enabled")
+		p.waitGroup.Add(1)
+		go func() {
+			defer p.waitGroup.Done()
+
+			ticker := time.NewTicker(time.Minute)
+			defer ticker.Stop()
+
+			db, err := p.client.Store.GetMasterDB()
+			if err != nil {
+				p.API.LogError("Could not initiate the database connection", "error", err.Error())
+				return
+			}
+			defer db.Close()
+			var currentList []*mmModel.ClusterDiscovery
+
+			for {
+				select {
+				case <-ticker.C:
+					list, err := pingClusterDiscoveryTable(db, p.client.Store.DriverName(), *p.API.GetConfig().ClusterSettings.ClusterName)
```

Comment:
@agnivade this is not the final version, but did you have something like this in your mind for getting the cluster nodes? Just wanted to clarify that before rectifying this code.

Reply:
Yeah seems like this is what we need to do. I initially thought we had some API to get all nodes and then a listener for any new node getting removed/added. But it looks like all we have is a listener for leaderChanged. So this is fine for now :+1:

Reply:
Perfect, I'll iterate from here then.